### PR TITLE
fix: make dev-server error message appear only once when CT is not configured

### LINKED ISSUE #17089: Double errors in component testing
### Current behavior
When starting up cypress component testing without updating pluginsFile, I get a very big error twice.

### Desired behavior
It should not display the error message twice
It should hide the full cypress stack trace

### Test code to reproduce
```sh
vue create new-app
cd new-app
yarn add -D cypress
yarn cypress open-ct
```

### Versions

Cypress 7.6.0

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes #17089

### User facing changelog

Simplified the error when the event `dev-server:start` is not configured and users start the component testing

### How has the user experience changed?

Before:

```log
Error: It is required to register dev-server plugin that implements `dev-server:start` event for component testing.
    at Object.start (/Users/bart/Documents/github/cypress/packages/server/lib/plugins/dev-server.js:36:13)
    at /Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:100:45
From previous event:
    at /Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:99:18
    at processImmediate (internal/timers.js:461:21)
From previous event:
    at ProjectCt._initPlugins (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:91:14)
    at onOpen (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:50:29)
    at callback (/Users/bart/Documents/github/cypress/node_modules/graceful-fs/polyfills.js:299:20)
    at callback (/Users/bart/Documents/github/cypress/packages/server/node_modules/graceful-fs/polyfills.js:289:20)
    at FSReqCallback.oncomplete (fs.js:184:5)
From previous event:
    at ProjectCt.open (/Users/bart/Documents/github/cypress/packages/server/lib/project-base.ts:124:14)
    at ProjectCt.open (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:47:22)
    at Object.create (/Users/bart/Documents/github/cypress/packages/server/lib/open_project.js:353:26)
    at /Users/bart/Documents/github/cypress/packages/server-ct/index.ts:66:39
    at processImmediate (internal/timers.js:461:21)
Error: It is required to register dev-server plugin that implements `dev-server:start` event for component testing.
    at Object.start (/Users/bart/Documents/github/cypress/packages/server/lib/plugins/dev-server.js:36:13)
    at /Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:100:45
From previous event:
    at /Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:99:18
    at processImmediate (internal/timers.js:461:21)
From previous event:
    at ProjectCt._initPlugins (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:91:14)
    at onOpen (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:50:29)
    at callback (/Users/bart/Documents/github/cypress/node_modules/graceful-fs/polyfills.js:299:20)
    at callback (/Users/bart/Documents/github/cypress/packages/server/node_modules/graceful-fs/polyfills.js:289:20)
    at FSReqCallback.oncomplete (fs.js:184:5)
From previous event:
    at ProjectCt.open (/Users/bart/Documents/github/cypress/packages/server/lib/project-base.ts:124:14)
    at ProjectCt.open (/Users/bart/Documents/github/cypress/packages/server-ct/src/project-ct.ts:47:22)
    at Object.create (/Users/bart/Documents/github/cypress/packages/server/lib/open_project.js:353:26)
    at /Users/bart/Documents/github/cypress/packages/server-ct/index.ts:66:39
    at processImmediate (internal/timers.js:461:21)
✨  Done in 26.45s.
✨  Done in 26.70s.
```

After:

```log
yarn workspace v1.22.10
yarn run v1.22.10
$ node ../../scripts/cypress.js open-ct --project ${PWD}
(node:62958) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `Cypress --trace-warnings ...` to show where the warning was created)
To run component-testing, cypress needs the `dev-server:start` event.

Implement it by adding a `on('dev-server:start', () => startDevServer())` call in your pluginsFile.

You can find the 'pluginsFile' at the following path:

/Users/AppleSeed/project/cypress/plugins/index.js

To learn more about how to set up component testing:

https://on.cypress.io/component-testing

✨  Done in 21.16s.
✨  Done in 21.41s.
```

### CODE REVIEW

File: packages/server/lib/errors.js [hunk 1]
```diff
@@ -965,6 +965,14 @@ const getMsgByType = function (type, arg1 = {}, arg2, arg3) {
 
         ${chalk.yellow(arg2)}
       `
+    case 'CT_NO_DEV_START_EVENT':
+      return stripIndent`\
```
Comment (lmiller1990) [inline packages/server/lib/errors.js:969]: nit: Can we not include the `\` in the message:

```
To run component-testing, cypress needs the \`dev-server:start\` event. 
````

The `\` look a bit weird.

File: packages/server/lib/plugins/dev-server.js [hunk 2]
```diff
@@ -32,8 +33,7 @@ const API = {
 
   start ({ specs, config }) {
     if (!plugins.has('dev-server:start')) {
-      // TODO: add link to the documentation in the error message
-      throw new Error('It is required to register dev-server plugin that implements `dev-server:start` event for component testing.')
+      return errors.throw('CT_NO_DEV_START_EVENT', config.pluginsFile)
```
Comment (lmiller1990) [inline packages/server/lib/plugins/dev-server.js:36]: Is the second arg (`config.pluginsFile`) needed? Seems it is not used.

Comment (elevatebart) [reply packages/server/lib/plugins/dev-server.js:36]: Good catch, my plan was to use it in the error to offer a clickable target for the user.

`Here is the path to your pluginsFile: /Users/dev/path`

File: packages/server/lib/errors.js (hunk 1 above)
Comment (elevatebart) [reply packages/server/lib/errors.js:969]: The backslash are not in the command message. They are only necessary for the backticks to be displayed in the final error.
My description was confusing.

(I did a poor Copy-Paste)

**Review by lmiller1990 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**